clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/KafkaStoreIngestionService.java

```diff
@@ -1054,6 +1079,73 @@ public AdminResponse getConsumptionSnapshots(String topicName, ComplementSet<Int
     return response;
   }
 
+  /**
+   * Return the metadata information for the given store. The data is retrieved from its respective repositories which
+   * originate from the VeniceServer.
+   * @param storeName
+   * @return {@link MetadataResponse} object that holds all the information required for answering a server metadata
+   * fetch request.
+   */
+  @Override
+  public MetadataResponse getMetadata(String storeName) {
+    MetadataResponse response = new MetadataResponse();
+    try {
+      Store store = metadataRepo.getStoreOrThrow(storeName);
+
+      Map<CharSequence, CharSequence> partitionerParams =
+          new HashMap<>(store.getPartitionerConfig().getPartitionerParams());
+      VersionProperties versionProperties = new VersionProperties(
+          store.getCurrentVersion(),
+          store.getVersion(store.getCurrentVersion()).get().getCompressionStrategy().getValue(),
+          store.getPartitionCount(),
```

Comment:
Sorry that I didn't make it clear in the previous comment but check the `Version` interface. We want to get these configs from `Version` as much as possible. Since the same problem applies with these other configs too like partition count and partitioner config. If we are getting it for `VersionProperties` for a specific version then we need to get it from `Version` and not `Store`